### PR TITLE
Add stream mode function to pipeline

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: lmdeploy/serve/async_engine.py [hunk 1]
```diff
@@ -314,6 +317,80 @@ async def gather():
...
+                                       EngineGenerationConfig]] = None,
+            do_preprocess: bool = True,
+            **kwargs):
+        """Inference a batch of prompts with stream mode.
+
+        Args:
+            prompts (List[str] | str): a batch of prompts
+            gen_config (GenerationConfig | None): a instance of
+                GenerationConfig. Default to None.
+            chat_template_config (ChatTemplateConfig | None):a instance of
```
Comment (RunningLeon) [inline lmdeploy/serve/async_engine.py:333]: no `chat_template_config` in the `stream_infer`

File: lmdeploy/serve/async_engine.py [hunk 2]
```diff
@@ -314,6 +317,80 @@ async def gather():
...
+                ChatTemplateConfig. Default to None.
+            do_preprocess (bool): whether pre-process the messages. Default to
+                True, which means chat_template will be applied.
+        """
+        need_list_wrap = isinstance(prompts, str) or isinstance(
```
Comment (RunningLeon) [inline lmdeploy/serve/async_engine.py:338]: type hint for `prompts`  is `Union[List[str], str]`, but here the type is `List[Dict]`

File: lmdeploy/serve/async_engine.py [hunk 3]
```diff
@@ -314,6 +317,80 @@ async def gather():
...
+            prompts[0], Dict)
```
Comment (RunningLeon) [inline lmdeploy/serve/async_engine.py:339]: type hint for `prompts`  is `Union[List[str], str]`, but here the type is `List[Dict]`

**Comment by shell-nlp** [comment]
一直所期待的流式方式。

**Review by RunningLeon — APPROVED** [review]
LGTM

File: lmdeploy/messages.py [hunk 4]
```diff
@@ -166,4 +166,5 @@ class Response:
     """Pack all response information together."""
     text: str
     generate_token_len: int
+    request_id: int
```
Comment (lvhan028) [inline lmdeploy/messages.py:169]: 建议使用 session_id，和之后讲interactive mode对照起来

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**